builtInFunctions/esdtDataStorage.go

```diff
@@ -411,8 +414,21 @@ func (e *esdtDataStorage) SaveESDTNFTToken(
 	return marshaledData, acnt.AccountDataHandler().SaveKeyValue(esdtNFTTokenKey, marshaledData)
 }
 
-// removeNFTMetadataFromSystemAccountIfNeeded removes the NFT metadata from the system account if needed
-func (e *esdtDataStorage) removeNFTMetadataFromSystemAccountIfNeeded(esdtTokenKey []byte, nonce uint64, esdtData *esdt.ESDigitalToken) error {
+func (e *esdtDataStorage) checkTheCorrectTokenTypeIsSet(esdtData *esdt.ESDigitalToken) bool {
+	if !e.enableEpochsHandler.IsFlagEnabled(DynamicEsdtFlag) {
+		return true
+	}
+
+	if esdtData.Type != uint32(core.NonFungible) {
```

Comment:
shouldn't we put these 2 checks together? if dynamic esdt flag is enabled it doesn't really mean that the token type was updated

Reply:
If the tokent type is `!= core.NonFungible` it means that for sure it was updated. I think it is more obvious and easy to understand as 2 separate checks.